src/components/DownloadComponents/hooks/useDownloadExistenceCheck.tsx

```diff
@@ -36,21 +36,28 @@ const useDownloadExistenceCheck = (props: UseDownloadExistenceCheckProps) => {
           await getDownloadResourcesExistence(
             lessonSlug,
             resourceTypesAsString,
-
             isLegacyDownload,
           );
 
-        const resourcesExistenceAsArray = resourceExistence
-          ? Object.entries(
-              resourceExistence as Partial<
-                Record<DownloadResourceType, boolean>
-              >,
-            )
-          : [];
+        console.log({ resourceExistence });
+
+        const resourcesExistenceAsArray: {
+          item: DownloadResourceType;
+          exists: boolean;
+        }[] = resourceExistence.map((r: unknown[]) => {
+          const k = r[0] as DownloadResourceType;
+          const v = r[1] as Record<string, unknown>;
```

Comment:
can we decode rather than cast?

Reply:
@kimon-satan if you merge main it to your branch all this stuff should now be typed courtesy of zod